### ISSUE TITLE
Shadow arbitrary scaling

### ISSUE THREAD

**Issue by stdedos** [Xpra-org/xpra#3140]
Continuing from https://github.com/Xpra-org/xpra/issues/2903, it may be a good time to handle arbitrary scaling correctly too

```
"Xpra-Python3-x86_64_4.2-r29129\xpra_cmd" shadow ssh://user@ip/1 --ssh="plink -ssh -agent" -d scaling --env=XPRA_SHADOW_REFRESH_DELAY=200 --title="@title@ on @@/@server-display@" --headerbar=off --opengl=no --bandwidth-limit=6Mbps --bandwidth-detection=no --desktop-scaling=0.75 --webcam=no --speaker=off --microphone=off --pulseaudio=no

args=--ssh="plink -ssh -agent" -d scaling --env=XPRA_SHADOW_REFRESH_DELAY=200 --title="@title@ on @@/@server-display@" --headerbar=off --opengl=no --bandwidth-limit=6Mbps --bandwidth-detection=no --desktop-scaling=0.75 --webcam=no --speaker=off --microphone=off --pulseaudio=no
XPRA_CUSTOM_TITLE_BAR=0
XPRA_EXECUTABLE=Xpra-Python3-x86_64_4.2-r29129
XPRA_NETWORK_ADAPTER_TYPE=wifi
XPRA_SCROLL_ENCODING=0

2021-05-27 11:50:22,040 Xpra GTK3 client version 4.2-r29129 (g5addf2823) 64-bit
2021-05-27 11:50:22,044  running on Microsoft Windows 10
2021-05-27 11:50:22,066 can_scale(0.75)=True
2021-05-27 11:50:22,070 parse_scaling(('0.75', 4160, 1440, 0.1, 8.0))
2021-05-27 11:50:22,090 parse_scaling(0.75) parsed items=(0.75, 0.75)
2021-05-27 11:50:22,092 parse_scaling(0.75)=(0.75, 0.75)
2021-05-27 11:50:22,092 scaling((0.75, 0.75))=(0.75, 0.75)
2021-05-27 11:50:24,938 GStreamer version 1.18.4 for Python 3.8.9 64-bit
2021-05-27 11:50:25,455 created named pipe 'Xpra\4576'
2021-05-27 11:50:25,866 keyboard layout code 0x409
2021-05-27 11:50:25,871 identified as 'United States - English' : us
2021-05-27 11:50:26,633  keyboard settings: layout=us
2021-05-27 11:50:26,716  desktop size is 4160x1440 with 1 screen:
2021-05-27 11:50:26,735   Default (1100x381 mm - DPI: 96x96) workarea: 4160x1400
2021-05-27 11:50:26,737     Generic PnP Monitor 1600x900 at 0x534 (309x174 mm - DPI: 132x131) workarea: 1600x860 at 0x534
2021-05-27 11:50:26,740     C32JG5x 2560x1440 at 1600x0 (697x392 mm - DPI: 93x93) workarea: 2560x1400 at 1600x0
2021-05-27 11:50:26,745  downscaled to 75%, virtual screen size: 5547x1920
2021-05-27 11:50:26,747   Default (1100x381 mm - DPI: 128x128) workarea: 5547x1867
2021-05-27 11:50:26,749     Generic PnP Monitor 2133x1200 at 0x712 (309x174 mm - DPI: 175x175) workarea: 2133x1147 at 0x712
2021-05-27 11:50:26,750     C32JG5x 3413x1920 at 2133x0 (697x392 mm - DPI: 124x124) workarea: 3413x1867 at 2133x0
2021-05-27 11:50:31,314 enabled remote logging
2021-05-27 11:50:31,319 Xpra shadow server version 4.3-r29154 64-bit
2021-05-27 11:50:31,320  running on Linux Ubuntu 20.04 focal
2021-05-27 11:50:31,321  remote desktop size is 6400x1440 with 1 screen:
2021-05-27 11:50:31,323   :1.0 (1505x342 mm - DPI: 108x107) workarea: 6334x1315 at 66x125
2021-05-27 11:50:31,324     DEL DP-0 2560x1440 (597x336 mm - DPI: 109x109) workarea: 2494x1413 at 66x27
2021-05-27 11:50:31,325     DEL DP-2 1920x1080 at 2560x98 (527x296 mm - DPI: 93x93) workarea: 1854x1053 at 2626x125
2021-05-27 11:50:31,326     DEL DP-4 1920x1080 at 4480x98 (527x296 mm - DPI: 93x93) workarea: 1854x1053 at 4546x125
2021-05-27 11:50:32,383 UI thread is now blocked
2021-05-27 11:50:32,830 UI thread is running again, resuming
2021-05-27 11:50:33,654 running, 3 windows

(xpra_cmd:4576): Pango-WARNING **: 11:50:34.113: couldn't load font "DejaVu Serif Not-Rotated 59.796875", falling back to "Sans Not-Rotated 59.796875", expect ugly output.

(xpra_cmd:4576): Pango-WARNING **: 11:50:34.138: couldn't load font "URW Bookman L Not-Rotated 52.76171875", falling back to "Sans Not-Rotated 52.76171875", expect ugly output.

(xpra_cmd:4576): Pango-WARNING **: 11:50:34.169: couldn't load font "Bitstream Vera Sans Not-Rotated 14.662109375", falling back to "Sans Not-Rotated 14.662109375", expect ugly output.
```
![Xpra_cmd_2021-05-27_11-51-03](https://user-images.githubusercontent.com/133706/119796904-733a0280-bee2-11eb-9f77-4bfefcc173f8.png)
![Xpra_cmd_2021-05-27_11-51-28](https://user-images.githubusercontent.com/133706/119796918-759c5c80-bee2-11eb-8f17-b1f1aa00e732.png)

**Comment by totaam** [comment]
Please try 28cd217 - circa r30322 or later.

**Comment by stdedos** [comment]
Latest available version for Ubuntu Focal is r30283 (76fc65878) 😕
Updating to that causes [redact-display-_1-20211101-130840.log](https://github.com/Xpra-org/xpra/files/7452659/redact-display-_1-20211101-130840.log), and no windows are rendered.

Unless you mean that this a client-only change.
... which ... SEEMS TO BE 😀

🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉

Verified with 28cd21705d116d1674ca5baaac54b0023d307d81 AKA r30322

**Comment by totaam** [comment]
> Updating to that causes ...

Oops, yes. Fixed days ago: 63777c94e59e63f25df26ec5330c6e1e17483adf

I'll spin up the buildbot now.
